app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/login/LoginViewModel.kt

```diff
@@ -1,6 +1,78 @@
 package com.github.alfabravo2013.readyforexams.presentation.login
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.alfabravo2013.readyforexams.R
+import com.github.alfabravo2013.readyforexams.repository.AuthenticationResult
+import com.github.alfabravo2013.readyforexams.repository.LoginRepositoryImpl
+import com.github.alfabravo2013.readyforexams.util.SingleLiveEvent
+import com.github.alfabravo2013.readyforexams.util.isInvalidEmail
+import com.github.alfabravo2013.readyforexams.util.isInvalidPassword
+import kotlinx.coroutines.launch
 
 class LoginViewModel : ViewModel() {
+    // temporary solution until DI is set up
+    private val authenticationApi = LoginRepositoryImpl
```

Comment:
loginRepository not authenticationApi. try to name the fields in the same way that you name the class.